### PR TITLE
Suggest content localisation #7669

### LINKED ISSUE #7669: Suggest content localisation
When user localises a content in a layer, show a dialog: "`Translate the content to <language of the layer>?`" (see conditions for showing the dialog below). If user answers "No", close the dialog and don't do anything (we may add "`Translate`" button for explicit translation later on). If "Yes", ask Assistant to translate texts in all text inputs and populate populate them with the results.

Conditions for showing the dialog (all of these should be met):
* AI Assistant is installed and active
* Content is in a layer
* Layer has a language
* User clicks "Localise"
* There's at least one text input (TextLine, TextArea or HtmlArea)
* At least one of these inputs is not empty

### PR DESCRIPTION
**Pull request by edloidas**
Moved AIAissistant wrapper to separate instance.
Updated AiAssistant events to make it always active, even before opened. 
Added checks to show confirmation dialog.

_Requires https://github.com/enonic/app-saga/pull/368_

### CODE REVIEW

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts [hunk 1]
```diff
@@ -2686,6 +2689,14 @@ export class ContentWizardPanel
         return this.splitPanel;
     }
 
+    isTranslatable(): boolean {
+        const content = this.getContent();
+
+        return AI.get().canTranslate() &&
+               (this.isContentExistsInParentProject() && content.hasOriginProject()) &&
+               !!ProjectContext.get().getProject().getLanguage();
+    }
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts:2698]: There's no check here for that the form only contains input types we can translate (TextLine, TextArea & HtmlArea). Or are we going to add this later?

Comment (alansemenov) [reply modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts:2698]: Nevermind that - the check is in Saga

File: modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-import {EditContentEvent} from '../../event/EditContentEvent';
-import {ContentSummaryAndCompareStatus} from '../../content/ContentSummaryAndCompareStatus';
-import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {DefaultErrorHandler} from '@enonic/lib-admin-ui/DefaultErrorHandler';
 import {showWarning} from '@enonic/lib-admin-ui/notify/MessageBus';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {ContentSummaryAndCompareStatus} from '../../content/ContentSummaryAndCompareStatus';
+import {EditContentEvent} from '../../event/EditContentEvent';
+import {ContentTreeGrid} from '../ContentTreeGrid';
+import {ContentsLocalizer} from './ContentsLocalizer';
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts]: Can we rename this to `ContentLocalizer` (single)? We usually don't use plural for content actions.

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts (hunk 1 above)
Comment (edloidas) [reply modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts:2698]: `AI.get().canTranslate()` do this, as it manages the translation itself.

File: modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-export class EditContentAction extends ContentTreeGridAction {
+export class EditContentAction
+    extends ContentTreeGridAction {
 
     private static MAX_ITEMS_TO_EDIT: number = 50;
 
     private isLocalize: boolean = false;
 
-    private contentsLocalizer?: ContentsLocalizer;
+    private contentsLocalizer?: ContentLocalizer;
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts]:
```suggestion
    private contentLocalizer?: ContentLocalizer;
```

File: modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts [hunk 4]
```diff
@@ -38,7 +38,7 @@ export class EditContentAction extends ContentTreeGridAction {
 
     private localizeContents(contents: ContentSummaryAndCompareStatus[]): void {
         if (!this.contentsLocalizer) {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts]:
```suggestion
        if (!this.contentLocalizer) {
```

File: modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts [hunk 5]
```diff
@@ -38,7 +38,7 @@ export class EditContentAction extends ContentTreeGridAction {
...
-            this.contentsLocalizer = new ContentsLocalizer();
+            this.contentsLocalizer = new ContentLocalizer();
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts]:
```suggestion
            this.contentLocalizer = new ContentLocalizer();
```

File: modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts [hunk 6]
```diff
@@ -38,7 +38,7 @@ export class EditContentAction extends ContentTreeGridAction {
...
         }
 
         this.contentsLocalizer.localizeAndEdit(contents).catch(DefaultErrorHandler.handle);
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/browse/action/EditContentAction.ts]:
```suggestion
        this.contentLocalizer.localizeAndEdit(contents).catch(DefaultErrorHandler.handle);
```

**Review by alansemenov — APPROVED** [review]

**Pull request merged**